src/include/processor/operator/copy/read_file.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "processor/operator/physical_operator.h"
+#include "storage/copier/table_copy_executor.h"
+
+namespace kuzu {
+namespace processor {
+
+class ReadFileMorsel {
+public:
+    ReadFileMorsel(common::offset_t nodeOffset, common::block_idx_t blockIdx, uint64_t numNodes,
+        std::string filePath)
+        : nodeOffset{nodeOffset}, blockIdx{blockIdx}, numNodes{numNodes}, filePath{std::move(
+                                                                              filePath)} {};
+
+    virtual ~ReadFileMorsel() = default;
+
+    static constexpr common::block_idx_t BLOCK_IDX_INVALID = UINT64_MAX;
+
+public:
+    common::offset_t nodeOffset;
+    common::block_idx_t blockIdx;
+    uint64_t numNodes;
+    std::string filePath;
+};
+
+class ReadFileSharedState {
+public:
+    explicit ReadFileSharedState(std::vector<std::string> filePaths)
+        : nodeOffset{0}, curBlockIdx{0}, filePaths{std::move(filePaths)}, curFileIdx{0}, numRows{
+                                                                                             0} {}
+
+    virtual ~ReadFileSharedState() = default;
+
+    virtual void countNumLines() = 0;
+
+    virtual std::unique_ptr<ReadFileMorsel> getMorsel() = 0;
+
+public:
+    uint64_t numRows;
+
+protected:
+    common::offset_t nodeOffset;
+    std::unordered_map<std::string, storage::TableCopyExecutor::FileBlockInfo> fileBlockInfos;
+    common::block_idx_t curBlockIdx;
+    std::vector<std::string> filePaths;
+    common::vector_idx_t curFileIdx;
+    std::mutex mtx;
+};
+
+class ReadFile : public PhysicalOperator {
+public:
+    ReadFile(DataPos fileRecordPos, std::shared_ptr<ReadFileSharedState> sharedState,
+        PhysicalOperatorType operatorType, uint32_t id, const std::string& paramsString)
+        : PhysicalOperator{operatorType, id, paramsString}, fileRecordPos{std::move(fileRecordPos)},
+          sharedState{std::move(sharedState)} {}
+
+    inline void initLocalStateInternal(ResultSet* resultSet, ExecutionContext* context) override {
+        fileRecordVector = resultSet->getValueVector(fileRecordPos).get();
+    }
+
+    inline void initGlobalStateInternal(kuzu::processor::ExecutionContext* context) override {
+        sharedState->countNumLines();
+    }
+
+    inline bool isSource() const override { return true; }
+
+    virtual std::shared_ptr<arrow::RecordBatch> readTuples(
```

Comment:
First of all, do we need to create RecordBatch object per function call? We seem to create a new RecordBatch and overwrite it in ValueVector per call. If so, shouldn't we always use the same RecordBatch object in value vector?

Also I wouldn't use unique_ptr as a parameter since there is no life cycle ownership changing